auv_sim/auv_sim_bridge/scripts/sim_torpedo_mock.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+
+import rospy
+import threading
+from std_srvs.srv import Trigger, TriggerResponse, TriggerRequest
+from gazebo_msgs.srv import SpawnModel, GetModelState, SetModelState
+from geometry_msgs.msg import Pose, Quaternion, Twist
+from tf.transformations import (
+    euler_from_quaternion,
+    quaternion_from_euler,
+    quaternion_matrix,
+)
+import rospkg
+import os
+import time
+
+
+def handle_torpedo_launch(req, torpedo_id):
```

Comment:
this seems to be a one giant function that handles some over complicated work. lets split  this?